native/src/main/java/io/ballerina/stdlib/http/api/HttpCallableUnitCallback.java

```diff
@@ -99,24 +112,14 @@ public void notifyFailure(BError error) { // handles panic and check_panic
     }
 
     private void sendFailureResponse(BError error) {
-        stopObservationWithContext();
+        stopObserverContext(true);
```

Comment:
@nadundesilva why do we ignore the `isManuallyClosed()` flag when there is an error ?

Reply:
This was the original behavior. Hence, I kept it as it is.

Reply:
The check on `observerContext.isManuallyClosed()` is also somewhat redundant right now though I think. Since we always set it to true. But it might be better to keep it, to avoid any possible issues later.

Reply:
Fixed with [64c98bfb0ab423a70c1c478489cb66043e01e392](https://github.com/ballerina-platform/module-ballerina-http/pull/807/commits/64c98bfb0ab423a70c1c478489cb66043e01e392)